src/Services/PatientService.php

```diff
@@ -44,10 +47,37 @@ public function validate($patient)
         $validator->required('sex')->lengthBetween(4, 30);
         $validator->required('DOB')->datetime('Y-m-d');
 
+        return $validator->validate($patient);
+    }
+
+    public function validateUpdate($pid, $patient)
+    {
+        $validator = new Validator();
+
+        $vPid = $this->validatePid($pid);
+        if ($vPid->isNotValid()) {
+            return $vPid;
+        }
+
+        $validator->optional('fname')->lengthBetween(2, 255);
+        $validator->optional('lname')->lengthBetween(2, 255);
+        $validator->optional('sex')->lengthBetween(4, 30);
+        if (key_exists('DOB', $patient)) {
+            $validator->required('DOB')->datetime('Y-m-d');
+        }
 
         return $validator->validate($patient);
     }
 
+    public function validatePid($pid)
+    {
+        $validator = new Validator();
+        $tmp['pid'] = $this->verifyPid($pid);
+        $validator->required('pid')->numeric();
+
+        return $validator->validate($tmp);
```

Comment:
If validation fails here, i.e the API tries to update a non existing user, I think the message will be, pid is required, or pid should be numeric, that could be confusing as the API user could have passed in a numeric value(satisfying but numeric and required condition), I think a better message should read, patient does not exist, is there anyway we can set our own custom error message or so?

Reply:
Yes, they give the option to customize message. 
http://validator.particle-php.com/en/latest/messages/

Reply:
Ya, they give option to overwrite. We need to overwrite some more places so better do that in a separate PR.
http://validator.particle-php.com/en/latest/messages/

Reply:
Good catch and I saw that but left for Yash as he is doing validation and may want to handle in a different way. I have no preference. Wait that's not true! My pref is I don't have to do it:)

Reply:
> satisfying but numeric and required condition)

Also note that pid is verified as a current patient or not. Issue is once it fails verify it returns null causing message to interface to be pid can't be null where it should be patient doesn't exist or such...